tests/v2/test_0835-datetime-type.py

```diff
@@ -47,454 +53,692 @@ def test_date_time_minmax():
+                np.datetime64("2020-01-27T10:41:11"),
+            ],
+        ]
+    )
+    assert ak.to_list(array) == [
+        [
+            np.datetime64("2020-03-27T10:41:11"),
+            np.datetime64("2020-01-27T10:41:11"),
+            np.datetime64("2020-05"),
+            np.datetime64("2020-01-27T10:41:11"),
+            np.datetime64("2020-04-27T10:41:11"),
+        ],
+        [
+            np.datetime64("2020-04-27"),
+            np.datetime64("2020-02-27T10:41:11"),
+            np.datetime64("2020-01-27T10:41:11"),
+            np.datetime64("2020-06-27T10:41:11"),
+        ],
+        [
+            np.datetime64("2020-02-27T10:41:11"),
+            np.datetime64("2020-03-27T10:41:11"),
+            np.datetime64("2020-01-27T10:41:11"),
+        ],
+    ]
+    assert ak._v2.operations.reducers.min(array) == np.datetime64("2020-01-27T10:41:11")
+    assert ak._v2.operations.reducers.max(array) == np.datetime64("2020-06-27T10:41:11")
+
+
+def test_highlevel_min_max():
+    array = ak.Array(
+        [
+            [
+                np.datetime64("2020-03-27T10:41:11"),
+                np.datetime64("2020-01-27T10:41:11"),
+                np.datetime64("2020-05"),
+                np.datetime64("2020-01-27T10:41:11"),
+                np.datetime64("2020-04-27T10:41:11"),
+            ],
+            [
+                np.datetime64("2020-04-27"),
+                np.datetime64("2020-02-27T10:41:11"),
+                np.datetime64("2020-01-27T10:41:11"),
+                np.datetime64("2020-06-27T10:41:11"),
+            ],
+            [
+                np.datetime64("2020-02-27T10:41:11"),
+                np.datetime64("2020-03-27T10:41:11"),
+                np.datetime64("2020-01-27T10:41:11"),
+            ],
+        ]
+    )
+    assert ak.to_list(array) == [
+        [
+            np.datetime64("2020-03-27T10:41:11"),
+            np.datetime64("2020-01-27T10:41:11"),
+            np.datetime64("2020-05"),
+            np.datetime64("2020-01-27T10:41:11"),
+            np.datetime64("2020-04-27T10:41:11"),
+        ],
+        [
+            np.datetime64("2020-04-27"),
+            np.datetime64("2020-02-27T10:41:11"),
+            np.datetime64("2020-01-27T10:41:11"),
+            np.datetime64("2020-06-27T10:41:11"),
+        ],
+        [
+            np.datetime64("2020-02-27T10:41:11"),
+            np.datetime64("2020-03-27T10:41:11"),
+            np.datetime64("2020-01-27T10:41:11"),
+        ],
+    ]
+    assert ak.to_list(ak._v2.operations.reducers.min(array, axis=0)) == [
+        np.datetime64("2020-02-27T10:41:11"),
+        np.datetime64("2020-01-27T10:41:11"),
+        np.datetime64("2020-01-27T10:41:11"),
+        np.datetime64("2020-01-27T10:41:11"),
+        np.datetime64("2020-04-27T10:41:11"),
+    ]
+    assert ak.to_list(ak._v2.operations.reducers.max(array, axis=0)) == [
+        datetime.datetime(2020, 4, 27, 0, 0),
+        datetime.datetime(2020, 3, 27, 10, 41, 11),
```

Comment:
something to look into: `max` returns `datetime.datetime` after update to master while `min` returns `numpy.datetime64`

Reply:
Note: I didn't look into this. It's still an issue.